SetupTemplate/Product.wxs

```diff
@@ -0,0 +1,135 @@
+           Language="1033"
+           Version="$(var.ProductVersion)"
+           Manufacturer="Cognite AS"
+           UpgradeCode="$(var.upgrade_guid)" >
+    <Package InstallerVersion="200" Compressed="yes" InstallScope="perMachine" />
+    <MajorUpgrade DowngradeErrorMessage="A newer version of $(var.NameSimple) is already installed." />
+    <MediaTemplate EmbedCab="yes" />
+
+    <UIRef Id="WixUI_InstallDir" />
+    <UIRef Id="WixUI_ErrorProgressText" />
+    <WixVariable Id="WixUILicenseRtf" Value="Resources\License.rtf" />
+    <WixVariable Id="WixUIBannerBmp" Value="Resources\InstBanner.bmp" />
+    <WixVariable Id="WixUIDialogBmp" Value="Resources\InstDialog.bmp" />
+    <WixVariable Id="WixUIExclamationIco" Value="Resources\black32x32.ico" />
+    <WixVariable Id="WixUIInfoIco" Value="Resources\black32x32.ico" />
+    <WixVariable Id="WixUINewIco" Value="Resources\black16x16.ico" />
+    <WixVariable Id="WixUIUpIco" Value="Resources\black16x16.ico" />
+
+    <Icon Id="AppIcon.ico" SourceFile="Resources\black16x16.ico" />
+
+    <Property Id="INSTALLFOLDER" Value="C:\Cognite">
+      <RegistrySearch Id="CogniteRegistry$(var.product_short_name)Folder" Type="raw" Root="HKLM" Key="Software\Cognite\$(var.product_short_name)" Name="InstallFolder" />
+    </Property>
+
+    <Property Id="WIXUI_INSTALLDIR" Value="INSTALLFOLDER" />
+    <Property Id="WIXUI_EXITDIALOGOPTIONALTEXT" Value="$(var.Name) - Is now installed." />
+
+    <Property Id="ARPCOMMENTS">$(var.Name)</Property>
+    <Property Id="ARPCONTACT">Cognite Support</Property>
+    <Property Id="ARPHELPLINK">http://www.cognite.com</Property>
+    <Property Id="ARPURLINFOABOUT">http://www.cognite.com</Property>
+    <Property Id="ARPURLUPDATEINFO">http://www.cognite.com</Property>
+    <Property Id="ARPPRODUCTICON">AppIcon.ico</Property>
+
+    <Feature Id="ProductFeature" Title="$(var.Name)" Level="1">
+      <ComponentGroupRef Id="ExecutableComponentGroup" />
+      <ComponentGroupRef Id="ConfigFilesComponentGroup" />
+      <!-- If you are referencing multiple separate products, add them here -->
+    </Feature>
+  </Product>
+
+  <Fragment>
+    <Directory Id="TARGETDIR" Name="SourceDir">
+      <Directory Id="ProgramFilesFolder">
+        <Directory Id="INSTALLFOLDER" Name="Cognite">
+          <Directory Id="ProductFolder" Name="$(var.product_short_name)">
+            <Directory Id="ConfigFolder" Name="config" />
+            <Directory Id="LogsFolder" Name="logs" />
+            <Directory Id="ExecutableFolder" Name="bin" />
+            <!-- Additional directories can be added here -->
+          </Directory>
+        </Directory>
+      </Directory>
+    </Directory>
+  </Fragment>
+
+  <Fragment>
+    <ComponentGroup Id="ExecutableComponentGroup">
+      <?if service='true' ?>
+        <Component Id="Primary_exe" Guid="*" Directory="ExecutableFolder">
+          <File Source="$(var.build_dir)\$(var.exe_name)" />
+
+          <?ifdef service_args?>
```

Comment:
Because compiletime conditionals are tags, they must be added outside of the entire tag you want to switch on. This is _not_ your average ifdef, the contents must be valid XML (which makes sense).

The consequence is that here where I just want to exclude a single parameter (to make it possible to not have service arguments), we have to replicate the entire tag containing the parameter, and all its contents.